### PR TITLE
Rename data to properties in _Object and DataObject

### PR DESCRIPTION
**Pull request by tsmith023**
This PR renames the `data` attribute within `_Object` and `DataObject` for clarity on its role in relation to the created Collection configurations. Since these map directly to the properties within the Class schema, naming it properties seems more appropriate.

### CODE REVIEW

**Review by dirkkul — APPROVED** [review]

**Comment by tsmith023** [comment]
@dirkkul, this will merge into [introduce_typed_dict_generic_to_collection](https://github.com/weaviate/weaviate-python-client/tree/introduce_typed_dict_generic_to_collection) so I won't merge it until that PR is also approved. Then they will cascade into `pydantic_experiment` 😁

**Comment by dirkkul** [comment]
can't we just merge it into pydantic_experiment?

**Comment by tsmith023** [comment]
It was tied up with the generics implementation but I've pulled that out too now

**Pull request closed without merging**
